app/controllers/autocomplete_controller.rb

```diff
@@ -0,0 +1,33 @@
+class AutocompleteController < ApplicationController
+  def github_repos
+    @repos = search_github_repos(params[:query])
+
+    respond_to do |format|
+      format.html { render partial: 'autocomplete/repository_suggestions', locals: { repos: @repos } }
+    end
+  end
+
+  private
+
+  def required_scopes
```

Comment:
@johndbritton 

Currently, there isn't any error handling in this controller (which is bad), I will try to add some of them to make it more robust.